### PR TITLE
NH-4535 add w3c style trace info to logs

### PR DESCRIPTION
**Pull request by prettycoder**
- rename traceId to TraceInfo
- rename and add fields for logged trace info
- extend CurrentTraceInfo SDK
- update tests
- refactor all occurencies of traceId in code base

NH-4535

### CODE REVIEW

File: lib/rails/generators/appoptics_apm/templates/appoptics_apm_initializer.rb [hunk 1]
```diff
@@ -128,14 +128,16 @@
   # Common Ruby and Rails loggers are auto-instrumented, so that they can include
   # the current Trace ID in log messages.
   #
-  # The added string will look like: "ao.traceId=7435A9FE510AE4533414D425DADF4E180D2B4E36-0"
+  # The added string will look like:
+  # "trace_id=7435a9fe510ae4533414d425dadf4e18 span_id=49e60702469db05f trace_flags=00"
   # It ends in '-1' if the request is sampled and in '-0' otherwise.
```
Comment (ronilan) [inline lib/rails/generators/appoptics_apm/templates/appoptics_apm_initializer.rb]: Delete comment on line 133.

File: lib/rails/generators/appoptics_apm/templates/appoptics_apm_initializer.rb [hunk 2]
```diff
@@ -128,14 +128,16 @@
   # Common Ruby and Rails loggers are auto-instrumented, so that they can include
   # the current Trace ID in log messages.
   #
-  # The added string will look like: "ao.traceId=7435A9FE510AE4533414D425DADF4E180D2B4E36-0"
+  # The added string will look like:
+  # "trace_id=7435a9fe510ae4533414d425dadf4e18 span_id=49e60702469db05f trace_flags=00"
```
Comment (ronilan) [inline lib/rails/generators/appoptics_apm/templates/appoptics_apm_initializer.rb:132]: 💯 Thanks for the clear 3-line spec of "Trace Context in Logs" . Been looking for something like that....

**Review by ronilan — DISMISSED** [review]

File: lib/rails/generators/appoptics_apm/templates/appoptics_apm_initializer.rb (hunk 1 above)
Comment (prettycoder) [reply lib/rails/generators/appoptics_apm/templates/appoptics_apm_initializer.rb]: right!

File: lib/rails/generators/appoptics_apm/templates/appoptics_apm_initializer.rb (hunk 2 above)
Comment (prettycoder) [reply lib/rails/generators/appoptics_apm/templates/appoptics_apm_initializer.rb:132]: :)

**Review by ronilan — APPROVED** [review]

**Pull request merged**